### PR TITLE
Update Lifecycle to 2.7.0. Use new LifecycleEventEffect.

### PR DESCRIPTION
**Pull request by dturner**
**What I have done and why**

- Updated AndroidX Lifecycle to version 2.7.0 ([release notes](https://developer.android.com/jetpack/androidx/releases/lifecycle#2.7.0)). 
- Replaced a `DisposableEffect` which was observing lifecycle events with the newly introduced `LifecycleEventEffect` ([docs](https://developer.android.com/topic/libraries/architecture/compose)) which does the same thing with less code. 

Before:
```
val lifecycleOwner = LocalLifecycleOwner.current
DisposableEffect(lifecycleOwner) {
    val observer = LifecycleEventObserver { _, event ->
       if (event == Lifecycle.Event.ON_STOP) {
            clearUndoState()
        }
    }
    lifecycleOwner.lifecycle.addObserver(observer)
    onDispose { lifecycleOwner.lifecycle.removeObserver(observer) }
}
```

After:

```
LifecycleEventEffect(Lifecycle.Event.ON_STOP) {
   clearUndoState()
}
```

Also added an instrumented test to verify the behavior.

### CODE REVIEW

File: feature/bookmarks/src/main/kotlin/com/google/samples/apps/nowinandroid/feature/bookmarks/BookmarksScreen.kt [hunk 1]
```diff
@@ -128,15 +126,8 @@ internal fun BookmarksScreen(
...
-    DisposableEffect(lifecycleOwner) {
-        val observer = LifecycleEventObserver { _, event ->
-            if (event == Lifecycle.Event.ON_STOP) {
-                clearUndoState()
-            }
-        }
-        lifecycleOwner.lifecycle.addObserver(observer)
-        onDispose { lifecycleOwner.lifecycle.removeObserver(observer) }
+    LifecycleEventEffect(Lifecycle.Event.ON_STOP) {
+        clearUndoState()
```
Comment (tunjid) [inline feature/bookmarks/src/main/kotlin/com/google/samples/apps/nowinandroid/feature/bookmarks/BookmarksScreen.kt:130]: niceeee

**Review by tunjid — APPROVED** [review]

**Review by hoc081098 — COMMENTED** [review]
Nice

File: feature/bookmarks/src/main/kotlin/com/google/samples/apps/nowinandroid/feature/bookmarks/BookmarksScreen.kt (hunk 1 above)
Comment (ianhanniballake) [reply feature/bookmarks/src/main/kotlin/com/google/samples/apps/nowinandroid/feature/bookmarks/BookmarksScreen.kt:130]: Looks like a great use case for this effect!

**Review by ianhanniballake — APPROVED** [review]

**Review by ianhanniballake — APPROVED** [review]
Looks like you will need to run `./gradlew :feature:bookmarks:spotlessApply` to fix spotless errors on `BookmarksScreenTest`

**Pull request merged**